src/bsp/src/posix/src/UserInterface.cpp

```diff
@@ -0,0 +1,50 @@
+#include "UserInterface.h"
+#include "ros/ros.h"
+#include <cstdarg>
+
+int UserInterface::print(const char* format, va_list args) const {
+    int ret = vprintf(format, args);
+    printf("\n\r");
+    return ret;
+}
+
+int UserInterface::print(const char* format, ...) const {
+    va_list args;
+    va_start(args, format);
+    int retValue = print(format, args);
+    va_end(args);
+
+    return retValue;
+}
+
+int UserInterface::logInfo(const char* format, va_list args) const {
+    const int bufferSize = 512;
+    char buffer[bufferSize];
+    int ret = vsnprintf(buffer, bufferSize, format, args);
+    ROS_INFO("%s", buffer);
+    return ret;
+}
+
+int UserInterface::logDebug(const char* format, va_list args) const {
+    const int bufferSize = 512;
+    char buffer[bufferSize];
+    int ret = vsnprintf(buffer, bufferSize, format, args);
+    ROS_DEBUG("%s", buffer);
+    return ret;
+};
+
+int UserInterface::logWarn(const char* format, va_list args) const {
+    const int bufferSize = 512;
+    char buffer[bufferSize];
+    int ret = vsnprintf(buffer, bufferSize, format, args);
+    ROS_WARN("%s", buffer);
+    return ret;
+};
+
+int UserInterface::logError(const char* format, va_list args) const {
+    const int bufferSize = 512;
+    char buffer[bufferSize];
+    int ret = vsnprintf(buffer, bufferSize, format, args);
```

Comment:
All the code with the buffer and vsnprintf is pretty redondant, could you make a private function that wraps all this instead of having it in all the functions (you could return a tuple of ret and a char pointer for example)?

Or at least use a constant for the bufferSize

Reply:
I meant a define of static variable used everywhere instead of redefining bufferSize everywhere.